crs_motion_planning/src/freespace_planning_server.cpp

```diff
@@ -0,0 +1,192 @@
+
+void tesseractRosutilsToMsg(trajectory_msgs::msg::JointTrajectory& traj_msg,
+                              const std::vector<std::string>& joint_names,
+                              const Eigen::Ref<const tesseract_common::TrajArray>& traj)
+{
+  assert(joint_names.size() == static_cast<unsigned>(traj.cols()));
+
+  // Initialze the whole traject with the current state.
+  std::map<std::string, int> jn_to_index;
+  traj_msg.joint_names.resize(joint_names.size());
+  traj_msg.points.resize(static_cast<size_t>(traj.rows()));
+
+  for (int i = 0; i < traj.rows(); ++i)
+  {
+    trajectory_msgs::msg::JointTrajectoryPoint jtp;
+    jtp.positions.resize(static_cast<size_t>(traj.cols()));
+
+    for (int j = 0; j < traj.cols(); ++j)
+    {
+      if (i == 0)
+        traj_msg.joint_names[static_cast<size_t>(j)] = joint_names[static_cast<size_t>(j)];
+
+      jtp.positions[static_cast<size_t>(j)] = traj(i, j);
+    }
+
+    jtp.time_from_start = rclcpp::Duration(i, 0);  // TODO: re-enable with rclcpp::Duration
+    traj_msg.points[static_cast<size_t>(i)] = jtp;
+  }
+}
+
+class PlanningServer: public rclcpp::Node
+{
+public:
+  PlanningServer()
+    : Node("planning_server_node")
+  {
+    plan_service_ = this->create_service<crs_msgs::srv::CallFreespaceMotion>("test_plan", std::bind(&PlanningServer::planService, this, std::placeholders::_1, std::placeholders::_2));
+    traj_publisher_ = this->create_publisher<trajectory_msgs::msg::JointTrajectory>("set_trajectory_test",10);
+    joint_state_listener_ = this->create_subscription<sensor_msgs::msg::JointState>("/joint_states", 2, std::bind(&PlanningServer::jointCallback, this, std::placeholders::_1));
+
+    joint_name_list_ = {"shoulder_pan_joint", "shoulder_lift_joint", "elbow_joint", "wrist_1_joint", "wrist_2_joint", "wrist_3_joint"};
+
+    std::string urdf_path, srdf_path;
+    urdf_path = ament_index_cpp::get_package_share_directory("crs_support") + "/urdf/crs.urdf";
+    srdf_path = ament_index_cpp::get_package_share_directory("crs_support") + "/urdf/ur10e_robot.srdf";
+    std::stringstream urdf_xml_string, srdf_xml_string;
+    std::ifstream urdf_in(urdf_path);
+    urdf_xml_string << urdf_in.rdbuf();
+    std::ifstream srdf_in(srdf_path);
+    srdf_xml_string << srdf_in.rdbuf();
+
+    tesseract_local_ = std::make_shared<tesseract::Tesseract>();
+    tesseract_scene_graph::ResourceLocator::Ptr locator = std::make_shared<tesseract_rosutils::ROSResourceLocator>();
+    tesseract_local_->init(urdf_xml_string.str(), srdf_xml_string.str(), locator);
+  }
+private:
+  void jointCallback(const sensor_msgs::msg::JointState::SharedPtr joint_msg)
+  {
+    curr_joint_state_ = *joint_msg;
+  }
+  void planService(std::shared_ptr<crs_msgs::srv::CallFreespaceMotion::Request> request,
+                    std::shared_ptr<crs_msgs::srv::CallFreespaceMotion::Response> response)
+  {
+    std::string target_frame = request->target_link;
+    std::string base_link_frame = "world";
```

Comment:
The "world" string should be in a global const variable or loaded from a parameters